tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts

```diff
@@ -0,0 +1,19 @@
+import { getBatches, getLatestBatch } from "@/api/batches";
+import { useQuery } from "@tanstack/react-query";
+import { pollingInterval } from "../lib/constants";
+
+export const useBatches = () => {
+  const { data: batches, isLoading: isBatchesLoading } = useQuery({
+    queryKey: ["batches"],
+    queryFn: () => getBatches(),
+    refetchInterval: pollingInterval,
```

Comment:
Not sure we're going to want this to always be polling btw. I changed it on main to only poll for the front page because we don't want the table page batches to constantly scroll off screen (remember they can come as fast as 1 per sec).

Reply:
Yeah, I used 5 seconds. However, we can take it out and have a refresh button on all the tables.

Reply:
Ok great, yeah we can figure that out when we sort the pagination and stuff.